lti_consumer/lti_xblock.py

```diff
@@ -1100,172 +1099,70 @@ def lti_launch_handler(self, request, suffix=''):  # pylint: disable=unused-argu
-            elif self.lti_advantage_deep_linking_enabled and 'deep_linking_content_launch' in lti_message_hint:
-                # Retrieve Deep Linking parameters using lti_message_hint parameter.
-                deep_linking_id = lti_message_hint.split(':')[1]
-                from lti_consumer.api import get_deep_linking_data  # pylint: disable=import-outside-toplevel
-                dl_params = get_deep_linking_data(deep_linking_id, block=self)
-
-                # Modify LTI launch and set ltiResourceLink parameters
-                lti_consumer.set_dl_content_launch_parameters(
-                    url=dl_params.get('url'),
-                    custom=dl_params.get('custom')
-                )
-
-            # Update context with LTI launch parameters
-            context.update({
-                "preflight_response": preflight_response,
-                "launch_request": lti_consumer.generate_launch_request(
-                    resource_link=str(self.location),  # pylint: disable=no-member
-                    preflight_response=preflight_response
-                )
-            })
-
-            template = loader.render_mako_template('/templates/html/lti_1p3_launch.html', context)
-            return Response(template, content_type='text/html')
-        except Lti1p3Exception as exc:
-            log.warning(
-                "Error preparing LTI 1.3 launch for block %r: %s",
-                str(self.location),  # pylint: disable=no-member
-                exc,
-            )
-            template = loader.render_django_template('/templates/html/lti_1p3_launch_error.html', context)
-            return Response(template, status=400, content_type='text/html')
-        except AssertionError as exc:
-            log.warning(
-                "Permission on LTI block %r denied for user %r: %s",
-                str(self.location),  # pylint: disable=no-member
-                self.external_user_id,
-                exc,
-            )
-            template = loader.render_django_template('/templates/html/lti_1p3_permission_error.html', context)
-            return Response(template, status=403, content_type='text/html')
+        # Backward compatibility - copy the config to the DB
+        self._save_lti_1p3_config_to_db()
+
+        from .plugin.views import launch_gate_endpoint  # pylint: disable=import-outside-toplevel
+        request.user = self.runtime.service(self, 'user').get_current_user()
+        return launch_gate_endpoint(request, suffix)
+
+    def _save_lti_1p3_config_to_db(self):
+        """
+        Save LTI 1.3 config to the database.
+        """
+        from .api import _get_lti_config  # pylint: disable=import-outside-toplevel
+        conf = _get_lti_config(block=self)
+        # Transfer the config to DB so that the django view can initialize the LTI1P3Consumer
+        # without having to load the XBlock
+        conf.lti_config = self._lti_1p3_config_as_dict()
```

Comment:
This was on oversight on my part. This will be replaced with `config.lti_config.update(self._lti_1p3_config_as_dict())` to preserve existing information.